ksql-engine/src/test/java/io/confluent/ksql/function/udf/array/EntriesTest.java

```diff
@@ -0,0 +1,134 @@
+  }
+
+  @Test
+  public void shouldComputeBigIntEntries() {
+    Map<String, Long> map = createMap(Long::valueOf);
+    shouldComputeEntries(map, () -> entriesUdf.entriesBigInt(map, false));
+  }
+
+  @Test
+  public void shouldComputeDoubleEntries() {
+    Map<String, Double> map = createMap(Double::valueOf);
+    shouldComputeEntries(map, () -> entriesUdf.entriesDouble(map, false));
+  }
+
+  @Test
+  public void shouldComputeBooleanEntries() {
+    Map<String, Boolean> map = createMap(i -> i % 2 == 0);
+    shouldComputeEntries(map, () -> entriesUdf.entriesBoolean(map, false));
+  }
+
+  @Test
+  public void shouldComputeStringEntries() {
+    Map<String, String> map = createMap(String::valueOf);
+    shouldComputeEntries(map, () -> entriesUdf.entriesString(map, false));
+  }
+
+  @Test
+  public void shouldComputeIntEntriesSorted() {
+    Map<String, Integer> map = createMap(i -> i);
+    shouldComputeEntriesSorted(map, () -> entriesUdf.entriesInt(map, true));
+  }
+
+  @Test
+  public void shouldComputeBigIntEntriesSorted() {
+    Map<String, Long> map = createMap(Long::valueOf);
+    shouldComputeEntriesSorted(map, () -> entriesUdf.entriesBigInt(map, true));
+  }
+
+  @Test
+  public void shouldComputeDoubleEntriesSorted() {
+    Map<String, Double> map = createMap(Double::valueOf);
+    shouldComputeEntriesSorted(map, () -> entriesUdf.entriesDouble(map, true));
+  }
+
+  @Test
+  public void shouldComputeBooleanEntriesSorted() {
+    Map<String, Boolean> map = createMap(i -> i % 2 == 0);
+    shouldComputeEntriesSorted(map, () -> entriesUdf.entriesBoolean(map, true));
+  }
+
+  @Test
+  public void shouldComputeStringEntriesSorted() {
+    Map<String, String> map = createMap(String::valueOf);
+    shouldComputeEntriesSorted(map, () -> entriesUdf.entriesString(map, true));
+  }
+
+  private <T> void shouldComputeEntries(
+      Map<String, T> map, Supplier<List<Struct>> supplier
+  ) {
+    List<Struct> out = supplier.get();
+    assertThat(out, hasSize(map.size()));
+    for (int i = 0; i < out.size(); i++) {
+      Struct struct = out.get(i);
+      T val = map.get(struct.getString("K"));
+      assertThat(val == null, is(false));
+      assertThat(val, is(struct.get("V")));
+    }
+  }
+
+  private <T> void shouldComputeEntriesSorted(Map<String, T> map, Supplier<List<Struct>> supplier) {
+    List<Struct> out = supplier.get();
+    List<Map.Entry<String, T>> entries = new ArrayList<>(map.entrySet());
+    entries.sort(Comparator.comparing(Entry::getKey));
+    assertThat(out.size(), is(entries.size()));
+    for (int i = 0; i < entries.size(); i++) {
+      Struct struct = out.get(i);
+      Map.Entry<String, T> entry = entries.get(i);
+      assertThat(struct.get("K"), is(entry.getKey()));
+      assertThat(struct.get("V"), is(entry.getValue()));
+    }
+  }
+
+  private <T> Map<String, T> createMap(Function<Integer, T> valueSupplier) {
+    Map<String, T> map = new HashMap<>();
+    for (int i = 0; i < ENTRIES; i++) {
```

Comment:
Add a test for null keys and null values. I see that you have an assertion for values to not be null but since the keys are optional schema, they could be null right? Also, there is not check in the actual class for null keys and values.

Reply:
fixed